clang/test/Sema/attr-nonblocking-constraints.cpp

```diff
@@ -420,19 +420,27 @@ class expected {
   constexpr expected()
     {}
 
+  // This is a deliberate corruption of the real implementation for simplicity.
   constexpr expected(const expected&)
+    noexcept(is_copy_constructible_v<_Tp> && is_copy_constructible_v<_Err>)
```

Comment:
I think something like this would be a simpler, more self-contained test:
```c++
constexpr bool foo() [[clang::nonblocking(false)]] { return true; }
void f() noexcept(foo()) [[clang::nonblocking]] {}
```
This currently causes a warning on godbolt.

Reply:
Speaking of constant-evaluation, this currently causes warnings:
```c++
constexpr bool foo() [[clang::nonblocking]] { 
    if consteval {
        auto p = new int;
        delete p;
    }

    return true; 
}
```
I think simply ignoring the body of a `consteval` if (or the else clause if there is one of a `!consteval` if) *should* be enough; that doesn’t have to be in this pr tho.